### PR TITLE
fix(NcInputField): Make sure label and placeholder is not jumping

### PR DESCRIPTION
**Pull request by susnux**
### ☑️ Resolves

Rework layout logic to ensure the label is not jumping on focus. This was caused as our border width changes when focusing.

#### Before

[vokoscreenNG-2024-07-18_02-44-45.webm](https://github.com/user-attachments/assets/a3666632-6ae2-4969-8e81-bc07ac3a251e)

#### After

[vokoscreenNG-2024-07-18_02-42-29.webm](https://github.com/user-attachments/assets/d6698d4d-3ba2-43b2-ac3f-19b11424d790)

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: src/components/NcInputField/NcInputField.vue [hunk 1]
```diff
@@ -440,31 +452,20 @@ export default {
...
-		}
 	}
 
 	&__input:focus + &__label,
 	&__input:not(:placeholder-shown) + &__label {
-		inset-block-start: -10px;
+		--input-label-font-size: 13px; // minimum allowed font size for accessibility
 		line-height: 1.5; // minimum allowed line height for accessibility
-		font-size: 13px; // minimum allowed font size for accessibility
+		inset-block-start: calc(-1.5 * var(--input-label-font-size) / 2);
```
Comment (Antreesy) [inline src/components/NcInputField/NcInputField.vue:461]: Feels like a magic number, TBH
```suggestion
		inset-block-start: calc(-0.75 * var(--input-label-font-size));
```

**Review by Antreesy — APPROVED** [review]
LGTM

**Review by marcoambrosini — APPROVED** [review]

File: src/components/NcInputField/NcInputField.vue (hunk 1 above)
Comment (susnux) [reply src/components/NcInputField/NcInputField.vue:461]: 1.5 is the line height so we get the real line height in pixels (1.5 * font size), to center we need the half ( thats the  / 2) and all of the needs to be negative as we move outside

**Pull request merged**